### PR TITLE
Upgrade makefile

### PR DESCRIPTION
**Pull request by Indraswara**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
+                    "description": "Get kernel symbols"
+                },
+                {
+                    "text": "set output-radix 16",
+                    "description": "Use hexadecimal output"
+                }
+            ],
+            "avoidWindowsConsoleRedirection": true
+        },
+    ]
```
Comment (Lock1) [inline .vscode/launch.json:45]: Please use LF for newline

File: makefile [hunk 2]
```diff
@@ -1,61 +1,90 @@
...
-
-%.o: %.cpp
-	@mkdir -p $(@D)
-	@$(CPPC) $(CFLAGS) $< -o $@
-	@echo Compiling $@...
-
-%.o: %.s
-	@mkdir -p $(@D)
-	@$(ASM) $(AFLAGS) $< -o $@
+#OBJECTS       = src/kernel.o src/kernel-entrypoint.o src/cpu/gdt.o
```
Comment (Lock1) [inline makefile:1]: Should've delete this unused code

File: makefile [hunk 3]
```diff
@@ -1,61 +1,90 @@
...
+# Compiler & linker
+ASM           = nasm
+LIN           = ld
+CPPC          = g++
+
+# Directory
+SOURCE_FOLDER = src
+# SOURCE_FOLDER_CODE = $(SOURCE_FOLDER)/code
+# SOURCE_FOLDER_ASM = $(SOURCE_FOLDER)/asm
+# SOURCE_FOLDER_KERNEL = $(SOURCE_FOLDER)/kernel
```
Comment (Lock1) [inline makefile:12]: Ditto

File: makefile [hunk 4]
```diff
@@ -1,61 +1,90 @@
...
+all: build
+build: iso
+clean:
+	rm -rf bin/
+	rm -rf $(OUTPUT_FOLDER)/*.iso
+	rm -rf $(OUTPUT_FOLDER)/kernel
+	rm -rf bin/*
+	rm -rf bin/*/
+
+kernel: $(OBJECTS) #TODO: I have to update $(OBJECTS)
```
Comment (Lock1) [inline makefile:60]: As `OBJECTS` already deleted, is this still needed?

**Review by Lock1 — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (Lock1) [reply .vscode/launch.json:45]: Also please ensure this PR only contain relevant changes

File: makefile (hunk 4 above)
Comment (Indraswara) [reply makefile:60]: which one? $(OBJECTS)? it still needed

gambar

Comment (Lock1) [reply makefile:60]: I see, I thought `OBJECTS` got removed completely

**Review by Lock1 — APPROVED** [review]

**Pull request merged**